src/Chunk.ts

```diff
@@ -68,8 +68,17 @@ function getGlobalName(
 	graph: Graph,
 	hasExports: boolean
 ) {
-	if (typeof globals === 'function') return globals(module.id);
-	if (globals) return globals[module.id];
+	let globalName: string | undefined;
+	if (typeof globals === 'function') {
+		globalName = globals(module.id);
+	} else if (globals) {
+		globalName = globals[module.id];
+	}
+
+	if (globalName) {
+		return globalName;
+	}
```

Comment:
I think it's ok to not warn when the user has not specified any configuration, as in many cases this is the required behaviour that the package name be the request package name (eg this is the default on npm that a request package name and output package name can match).

The issue maybe arises in environments in which you are defining a global function or map that is trying to be authoritative and then has these invisible misses.

Instead, perhaps let's only warn for the case where `globals` is provided but has misses, by adding that as another case above?

Reply:
I am not 100% sure about this, as for UMD, I noticed that a lot of guessed name would be incorrect. If for instance having a `.` in the name. Or when the umd bundle name is totally different, and without a warning, it's hard to rollup users to know that something might be wrong.

However I can do what you suggested.

Reply:
@guybedford pushed an update.